test/e2e/cross_os/test_job_submissions.py

```diff
@@ -632,47 +630,195 @@ def test_worker_uses_job_attachment_configuration(
+                                            "data": hash_string_script,
+                                            **(
+                                                {"filename": "hashscript.ps1"}
+                                                if os.environ["OPERATING_SYSTEM"] == "windows"
+                                                else {}
+                                            ),
+                                        }
+                                    ],
+                                },
+                            }
+                        ],
+                    }
+                )
+            )
+
+        config = configparser.ConfigParser()
+
+        set_setting("defaults.farm_id", deadline_resources.farm.id, config)
+        set_setting("defaults.queue_id", deadline_resources.queue_a.id, config)
+
+        job_id: Optional[str] = api.create_job_from_job_bundle(
+            job_bundle_path,
+            job_parameters,
+            priority=99,
+            max_retries_per_task=0,
+            config=config,
+            queue_parameter_definitions=[],
+        )
+        assert job_id is not None
+
+        job_details = Job.get_job_details(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            job_id=job_id,
+        )
+        job = Job(
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            template={},
+            **job_details,
+        )
+
+        # Query the session to check for progress percentage
+        complete_percentage: float = 0
+        while complete_percentage < 100:
+            sessions = deadline_client.list_sessions(
+                farmId=job.farm.id, queueId=job.queue.id, jobId=job.id
+            ).get("sessions")
+
+            if sessions:
+                session_actions = deadline_client.list_session_actions(
+                    farmId=job.farm.id,
+                    queueId=job.queue.id,
+                    jobId=job.id,
+                    sessionId=sessions[0]["sessionId"],
+                ).get("sessionActions")
+
+                for session_action in session_actions:
+                    if "syncInputJobAttachments" in session_action["definition"]:
+                        assert complete_percentage <= session_action["progressPercent"]
+                        complete_percentage = session_action["progressPercent"]
+
+        output_path: dict[str, list[str]] = get_job_output(
+            job=job, deadline_client=deadline_client, deadline_resources=deadline_resources
+        )
+
+        with (
+            open(os.path.join(list(output_path.keys())[0], "output_file.txt"), "r") as output_file,
+        ):
+            output_file_content = output_file.read()
+            # Verify that the hash is the same
+            assert output_file_content == combined_hash
+
+        shutil.rmtree(job_bundle_path)
```

Comment:
This is bad, it will only run when the test is finished successfully. We want to cleanup the job bundle no matter what, so we should either use a try catch or temporary directories.